src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java

```diff
@@ -0,0 +1,158 @@
+import java.awt.*;
+
+public class AdvancedSearchPanel extends JPanel {
+    private JTextField textFieldVideoBitrate;
+    private JTextField textFieldVideoFramerate;
+    private JTextField textFieldAudioBitrate;
+
+    public AdvancedSearchPanel() {
+        GridBagLayout gridBagLayout = new GridBagLayout();
+        gridBagLayout.columnWidths = new int[]{66, 0, 0};
+        gridBagLayout.rowHeights = new int[]{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+        gridBagLayout.columnWeights = new double[]{0.0, 1.0, Double.MIN_VALUE};
+        gridBagLayout.rowWeights = new double[]{0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, Double.MIN_VALUE};
+        setLayout(gridBagLayout);
+
+        JLabel lblVideo = new JLabel("Video");
+        lblVideo.setFont(new Font("Tahoma", Font.BOLD, 14));
+        GridBagConstraints gbc_lblVideo = new GridBagConstraints();
+        gbc_lblVideo.gridwidth = 2;
+        gbc_lblVideo.insets = new Insets(0, 0, 5, 0);
+        gbc_lblVideo.gridx = 0;
+        gbc_lblVideo.gridy = 0;
+        add(lblVideo, gbc_lblVideo);
+
+        JLabel lblCodecVideo = new JLabel("Codec: ");
+        GridBagConstraints gbc_lblCodecVideo = new GridBagConstraints();
+        gbc_lblCodecVideo.anchor = GridBagConstraints.EAST;
+        gbc_lblCodecVideo.insets = new Insets(0, 0, 5, 5);
+        gbc_lblCodecVideo.gridx = 0;
+        gbc_lblCodecVideo.gridy = 1;
+        add(lblCodecVideo, gbc_lblCodecVideo);
+
+        JComboBox comboBoxVideoCodec = new JComboBox();
+        GridBagConstraints gbc_comboBoxVideoCodec = new GridBagConstraints();
+        gbc_comboBoxVideoCodec.insets = new Insets(0, 0, 5, 0);
+        gbc_comboBoxVideoCodec.fill = GridBagConstraints.HORIZONTAL;
+        gbc_comboBoxVideoCodec.gridx = 1;
+        gbc_comboBoxVideoCodec.gridy = 1;
+        add(comboBoxVideoCodec, gbc_comboBoxVideoCodec);
+
+        JLabel lblBitrateVideo = new JLabel("Bitrate: ");
+        GridBagConstraints gbc_lblBitrateVideo = new GridBagConstraints();
+        gbc_lblBitrateVideo.anchor = GridBagConstraints.EAST;
+        gbc_lblBitrateVideo.insets = new Insets(0, 0, 5, 5);
+        gbc_lblBitrateVideo.gridx = 0;
+        gbc_lblBitrateVideo.gridy = 2;
+        add(lblBitrateVideo, gbc_lblBitrateVideo);
+
+        textFieldVideoBitrate = new JTextField();
+        GridBagConstraints gbc_textFieldVideoBitrate = new GridBagConstraints();
+        gbc_textFieldVideoBitrate.insets = new Insets(0, 0, 5, 0);
+        gbc_textFieldVideoBitrate.fill = GridBagConstraints.HORIZONTAL;
+        gbc_textFieldVideoBitrate.gridx = 1;
+        gbc_textFieldVideoBitrate.gridy = 2;
+        add(textFieldVideoBitrate, gbc_textFieldVideoBitrate);
+        textFieldVideoBitrate.setColumns(10);
+
+        JLabel lblFramerate = new JLabel("Framerate: ");
+        GridBagConstraints gbc_lblFramerate = new GridBagConstraints();
+        gbc_lblFramerate.anchor = GridBagConstraints.EAST;
+        gbc_lblFramerate.insets = new Insets(0, 0, 5, 5);
```

Comment:
Magical numbers.

Reply:
Numbers of coordinates in GridBagLayout, please refactor in the next Sprint